File: 2018-2019/zima/PF/lista2/zad1.ml
let rec sublists xs =
    let rec map_append ?(acc=[]) elem xss =
        match xss with
        | []    -> acc
        | xs::xss -> 
            let new_acc = xs :: (elem :: xs) :: acc
                in map_append ~acc:new_acc elem xss
    in match xs with
    | []    -> [[]]
    | y::ys -> 
        let ls = sublists ys
            in map_append y ls

File: 2018-2019/lato/Prolog/L4/zad1.pl
revall(Xss, Yss) :-
    is_list(Xss),
    !,
    revall(Xss, Yss, []).
revall(X, X).

revall([], Acc, Acc) :-
    !.
revall([Xs | Xss], Yss, Acc) :-
    revall(Xs, Zs),
    revall(Xss, Yss, [Zs | Acc]),
    !.
